### PR TITLE
RCv3 models for converegence

### PR DESCRIPTION
**Pull request by cyli**
This pull request addresses part of #821 and does the following:

- Add the RCv3 ILBDescription and ILBNode models
- Modifies the generic lb add and remove steps to produce `BulkAddToRCv3` and `BulkRemoveFromRCv3` steps
- Add RCv3 bits to the existing converge with LB tests, and adds some new tests
- Refactors some of the converge with LB tests

Still needed after this PR to complete #821 is to add RCv3 to gathering.

### CODE REVIEW

**Comment by cyli** [comment]
jenkins please run integration tests

**Comment by cyli** [comment]
jenkins please run integration tests

File: otter/test/convergence/test_planning.py [hunk 1]
```diff
@@ -31,13 +33,77 @@
     SetMetadataItemOnServer)
 
 
+def copy_clbdesc(clb_desc, condition=CLBNodeCondition.ENABLED, weight=1):
```
Comment (lvh) [inline otter/test/convergence/test_planning.py]: Nitpick: missing _ between clb and desc?

File: otter/test/convergence/test_planning.py [hunk 2]
```diff
@@ -31,13 +33,77 @@
...
+    """
+    Produce a :class:`CLBDescription` from another, but with the given
```
Comment (lvh) [inline otter/test/convergence/test_planning.py]: Just the condition is changed, not a given condition, right? That makes it sound like there's multiple conditions and you're specifying which one.

File: otter/convergence/planning.py [hunk 3]
```diff
@@ -216,7 +220,7 @@ def drain_and_delete_a_server(server):
     delete_error_steps = (
         [DeleteServer(server_id=server.id) for server in servers_in_error] +
         [RemoveNodesFromCLB(lb_id=lb_node.description.lb_id,
-                            node_ids=(lb_node.node_id,))
+                            node_ids=pset([lb_node.node_id]))
```
Comment (lvh) [inline otter/convergence/planning.py]: If the code previously accepted either, and that's a defect, there should probably be a typecheck for `PSet` in `RemoveNodesFromCLB`.

**Comment by lvh** [comment]
LGTM modulo comments; super thorough; thanks for working on this!

File: otter/convergence/planning.py (hunk 3 above)
Comment (cyli) [inline otter/convergence/planning.py]: https://github.com/rackerlabs/otter/issues/1151?

Comment (lvh) [inline otter/convergence/planning.py]: LGTM, thanks!

**Pull request merged**